app/src/main/java/com/jerboa/ui/components/registrationapplications/RegistrationApplications.kt

```diff
@@ -0,0 +1,496 @@
+}
+
+@OptIn(ExperimentalMaterialApi::class)
+@Composable
+fun RegistrationApplications(
+    appState: JerboaAppState,
+    registrationApplicationsViewModel: RegistrationApplicationsViewModel,
+    siteViewModel: SiteViewModel,
+    ctx: Context,
+    account: Account,
+    scope: CoroutineScope,
+    snackbarHostState: SnackbarHostState,
+    padding: PaddingValues,
+) {
+    val listState = rememberLazyListState()
+
+    // observer when reached end of list
+    val endOfListReached by remember {
+        derivedStateOf {
+            listState.isScrolledToEnd()
+        }
+    }
+
+    // act when end of list reached
+    if (endOfListReached) {
+        LaunchedEffect(Unit) {
+            account.doIfReadyElseDisplayInfo(
+                appState,
+                ctx,
+                snackbarHostState,
+                scope,
+                siteViewModel,
+            ) {
+                registrationApplicationsViewModel.appendApplications()
+            }
+        }
+    }
+
+    val refreshing = registrationApplicationsViewModel.applicationsRes.isRefreshing()
+
+    val refreshState =
+        rememberPullRefreshState(
+            refreshing = refreshing,
+            onRefresh = {
+                account.doIfReadyElseDisplayInfo(
+                    appState,
+                    ctx,
+                    snackbarHostState,
+                    scope,
+                    siteViewModel,
+                ) {
+                    registrationApplicationsViewModel.resetPage()
+                    registrationApplicationsViewModel.listApplications(
+                        registrationApplicationsViewModel.getFormApplications(),
+                        ApiState.Refreshing,
+                    )
+                    siteViewModel.fetchUnreadAppCount()
+                }
+            },
+        )
+
+    Column(
+        modifier = Modifier
+            .padding(padding),
+    ) {
+        Box(
+            modifier = Modifier.pullRefresh(refreshState),
+        ) {
+            JerboaPullRefreshIndicator(
+                refreshing,
+                refreshState,
+                Modifier
+                    .align(Alignment.TopCenter)
+                    .zIndex(100F),
+            )
+
+            if (registrationApplicationsViewModel.applicationsRes.isLoading()) {
+                LoadingBar()
+            }
+            when (val appsRes = registrationApplicationsViewModel.applicationsRes) {
+                ApiState.Empty -> ApiEmptyText()
+                is ApiState.Failure -> ApiErrorText(appsRes.msg)
+                is ApiState.Holder -> {
+                    val apps = appsRes.data.registration_applications
+                    LazyColumn(
+                        state = listState,
+                        modifier =
+                            Modifier
+                                .simpleVerticalScrollbar(listState)
+                                .fillMaxSize()
+                                .imePadding(),
```

Comment:
I'm pretty unsure where this imePadding should live, but it seems to work here.

Its necessary because each registration application form has an expandable `deny_reason` markdown field.